### PR TITLE
EVA-2623: Variant history

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java [hunk 1]
```diff
@@ -110,6 +115,37 @@ public ClusteredVariantsRestController(
...
+
+        if (allVariants.isEmpty() && allOperations.isEmpty()) {
+            throw new AccessionDoesNotExistException(identifier);
+        }
+        return ResponseEntity.ok(new VariantHistory<>(allVariants, allOperations));
+    }
+
+    public HistoryEventDTO<Long, ClusteredVariant> toHistoryEventDTO(IEvent<? extends IClusteredVariant, Long> operation) {
+        HistoryEvent<IClusteredVariant, Long> historyEvent = new HistoryEvent<IClusteredVariant, Long>(operation.getEventType(), operation.getAccession(),
+                operation.getInactiveObjects().get(0).getVersion(), operation.getResultInto(), operation.getCreatedDate(), operation.getInactiveObjects().get(0).getModel());
```
Comment (andresfsilva) [inline eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java]:
```suggestion
                operation.getInactiveObjects().get(0).getVersion(), operation.getDestinationAccession(), operation.getCreatedDate(), operation.getInactiveObjects().get(0).getModel());
```

Comment (andresfsilva) [inline eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java]: Please mind the lines length, it makes it easier to review it they don't exceed 120 characters

**Review by andresfsilva — CHANGES_REQUESTED** [review]

File: eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java (hunk 1 above)
Comment (nitin-ebi) [reply eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java]: this has been updated and pushed

Comment (nitin-ebi) [reply eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java]: sorry, looks like formatting issue again. Will correct

File: eva-accession-ws/pom.xml [hunk 2]
```diff
@@ -57,6 +57,17 @@
             <groupId>org.hsqldb</groupId>
             <artifactId>hsqldb</artifactId>
         </dependency>
+        <dependency>
+            <groupId>uk.ac.ebi.eva</groupId>
+            <artifactId>eva-accession-clustering</artifactId>
```
Comment (sundarvenkata-EBI) [inline eva-accession-ws/pom.xml:62]: Why is this dependency needed?

File: eva-accession-ws/src/test/java/uk/ac/ebi/eva/accession/ws/ClusteredVariantHistoryEndPointTest.java [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+    private void splitRS(List<SubmittedVariantEntity> submittedVariantEntities) throws Exception {
+        SubmittedVariantOperationEntity splitOperation = new SubmittedVariantOperationEntity();
+        splitOperation.fill(RSMergeAndSplitCandidatesReaderConfiguration.SPLIT_CANDIDATES_EVENT_TYPE,
+                submittedVariantEntities.get(0).getAccession(), "Hash mismatch with " +
+                        submittedVariantEntities.get(0).getClusteredVariantAccession(),
+                submittedVariantEntities.stream().map(SubmittedVariantInactiveEntity::new)
+                        .collect(Collectors.toList())
+        );
+        mongoTemplate.insert(Collections.singletonList(splitOperation), SubmittedVariantOperationEntity.class);
+        rsSplitWriter.write(Collections.singletonList(splitOperation));
```
Comment (sundarvenkata-EBI) [inline eva-accession-ws/src/test/java/uk/ac/ebi/eva/accession/ws/ClusteredVariantHistoryEndPointTest.java:416]: Judgment call: I am not sure explicitly invoking split writer (thereby creating a dependency on the clustering module) is a good idea. We know the end result of splits and merges. In my opinion, we should try to emulate that end result in data rather.

**Review by sundarvenkata-EBI — APPROVED** [review]

**Review by tcezard — APPROVED** [review]

**Pull request merged**